internal/credit/postgresadapter/ent/schema/balance_snapshot.go

```diff
@@ -0,0 +1,56 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/dialect"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/internal/credit"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type BalanceSnapshot struct {
+	ent.Schema
+}
+
+// Mixin of the CreditGrant.
+func (BalanceSnapshot) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{}, // maybe we don't need this
+	}
+}
+
+// Fields of the CreditGrant.
+func (BalanceSnapshot) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("owner_id").GoType(credit.GrantOwner("")).Immutable().SchemaType(map[string]string{
```

Comment:
why is it owner? is it a subject?

Reply:
for current usecases it transaltes to an entitlement, see the comment above for details

Reply:
The problem is that this way we loose the foreign key.
I'm wondering that evenn if connector is generic PG adapter should be specific to use-case.

Reply:
thats not a bad option, we can do that 👍  
i'm personally not that worried about missing the FKeys, I thought about it a bit when deciding to do it this way and couldn't really find a scenario where it would bite us